design-system/react/src/components/PasswordStrength/PasswordStrength.tsx

```diff
@@ -0,0 +1,114 @@
+import { Check, X } from 'phosphor-react';
+import { useMemo } from 'react';
+
+import { Box } from '../Box';
+import { Flex } from '../Flex';
+import { Heading } from '../Heading';
+import { Icon } from '../Icon';
+import type { PopoverProps } from '../Popover';
+import { Popover } from '../Popover';
+import { Text } from '../Text';
+
+import { PasswordDictionary } from './constants';
+import { passwordStrengthStyles } from './styles';
+import { passwordChecker, passwordStrengthCalculator } from './utils';
+
+import { createComponent } from '~/utils';
+
+export type PasswordStrengthProps = {
+  password: string;
+  minLength?: number;
+} & Omit<PopoverProps, 'content'>;
+
+export const PasswordStrength = createComponent<PasswordStrengthProps>(
+  ({ password, children, minLength = 6, ...props }) => {
+    const strength = useMemo(
```

Comment:
You can add this both helpers inside a hook, and use `useMemo` for the `passwordChecker` as well